tests/waku_relay/test_protocol.nim

```diff
@@ -0,0 +1,1302 @@
+      let msg4 = fakeWakuMessage(testMessage, pubsubTopic)
+      discard await otherNode.publish(pubsubTopic, msg4)
+      
+      # Then the message is received in both nodes
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        await otherHandlerFuture.withTimeout(3.seconds)
+        (pubsubTopic, msg4) == handlerFuture.read()
+        (pubsubTopic, msg4) == otherHandlerFuture.read()
+
+      # Finally stop the other node
+      await allFutures(otherSwitch.stop(), otherNode.stop())
+
+    xasyncTest "Relay can receive messages after subscribing and stopping without unsubscribing":
+      # Given a second node connected to the first one
+      let
+        otherSwitch = newTestSwitch()
+        otherPeerManager = PeerManager.new(otherSwitch)
+        otherNode = await newTestWakuRelay(otherSwitch)
+
+      await allFutures(otherSwitch.start(), otherNode.start())
+      let 
+        otherRemotePeerInfo = otherSwitch.peerInfo.toRemotePeerInfo()
+        otherPeerId = otherRemotePeerInfo.peerId
+
+      check await peerManager.connectRelay(otherRemotePeerInfo)
+
+      # Given both are subscribed to the same pubsub topic
+      var otherHandlerFuture = newPushHandlerFuture()
+      proc otherSimpleFutureHandler(topic: PubsubTopic, message: WakuMessage) {.async, gcsafe.} =
+        otherHandlerFuture.complete((topic, message))
+
+      discard otherNode.subscribe(pubsubTopic, otherSimpleFutureHandler)
+      discard node.subscribe(pubsubTopic, simpleFutureHandler)
+      check:
+        node.subscribedTopics == pubsubTopicSeq
+        otherNode.subscribedTopics == pubsubTopicSeq
+      
+      await sleepAsync(500.millis)
+
+      # FIXME: Inconsistent behaviour with Filter protocol.
+      # Given other node is stopped without unsubscribing
+      await allFutures(otherSwitch.stop(), otherNode.stop())
+
+      # When sending a message from node
+      let msg1 = fakeWakuMessage(testMessage, pubsubTopic)
+      discard await node.publish(pubsubTopic, msg1)
+
+      # Then the message is received in both nodes
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        await otherHandlerFuture.withTimeout(3.seconds)
+        (pubsubTopic, msg1) == handlerFuture.read()
+        (pubsubTopic, msg1) == otherHandlerFuture.read()
+      
+      # When sending a message from other node
+      handlerFuture = newPushHandlerFuture()
+      otherHandlerFuture = newPushHandlerFuture()
+      let msg2 = fakeWakuMessage(testMessage, pubsubTopic)
+      discard await otherNode.publish(pubsubTopic, msg2)
+      
+      # Then the message is received in both nodes
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        await otherHandlerFuture.withTimeout(3.seconds)
+        (pubsubTopic, msg2) == handlerFuture.read()
+        (pubsubTopic, msg2) == otherHandlerFuture.read()
+      
+      # Finally stop the other node
+      await allFutures(otherSwitch.stop(), otherNode.stop())
```

Comment:
I think otherSwitch and otherNode have already been stopped in line 1275

Reply:
Oh, yeah! Did that out of custom